cmd/skopeo/completions.go

```diff
@@ -4,10 +4,47 @@ import (
 	"github.com/containers/image/v5/tarball"
 	"github.com/containers/image/v5/transports"
 	"github.com/spf13/cobra"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
 )
 
-// autocompleteSupportedTransports list all supported transports with the colon suffix.
-func autocompleteSupportedTransports(cmd *cobra.Command, args []string, toComplete string) ([]cobra.Completion, cobra.ShellCompDirective) {
+func autocompleteTransports(cmd *cobra.Command, args []string, toComplete string) ([]cobra.Completion, cobra.ShellCompDirective) {
+	directive := cobra.ShellCompDirectiveNoSpace
+	// We still don't have a transport
+	if !strings.Contains(toComplete, ":") {
+		transports := supportedTransportSuggestions()
+		return transports, directive | cobra.ShellCompDirectiveNoFileComp
+	}
+	if strings.HasPrefix(toComplete, "oci-archive:") || strings.HasPrefix(toComplete, "docker-archive:") || strings.HasPrefix(toComplete, "sif:") || strings.HasPrefix(toComplete, "oci:") {
+		return nil, directive
+	}
+	if toComplete == "docker:" {
+		return []cobra.Completion{"docker://"}, directive | cobra.ShellCompDirectiveNoFileComp
+	}
+	if strings.HasPrefix(toComplete, "dir:") {
```

Comment:
here you then should match the transpart names as defined by the code and not hard code the strings
`docker.Transport.Name()`, `directory.Transport.Name()`,etc...